app/api/services/ocr/OcrManager.ts

```diff
@@ -0,0 +1,233 @@
+};
+
+const processResults = async (message: ResultsMessage): Promise<void> => {
+  await tenants.run(async () => {
+    try {
+      const [originalFile] = await files.get({ filename: message.params!.filename });
+      const [record] = await getForSourceFile(originalFile);
+
+      if (!record) return;
+
+      if (!message.success) {
+        await handleOcrError(record, originalFile, message);
+        return;
+      }
+
+      await processFiles(record, message, originalFile);
+      emitToTenant(message.tenant, 'ocr:ready', originalFile._id.toHexString());
+    } catch (e) {
+      handleError(e);
+    }
+  }, message.tenant);
+};
+
+const validateLanguage = async (language: string, ocrSettings?: { url: string }) => {
+  const _ocrSettings = ocrSettings || (await getSettings());
+  const supportedLanguages = await fetchSupportedLanguages(_ocrSettings);
+  return supportedLanguages.includes(LANGUAGES_MAP[language]);
+};
+
+const getStatus = async (file: EnforcedWithId<FileType>) => {
+  const [record] = await getForSourceOrTargetFile(file);
+
+  const status = record ? record.status : OcrStatus.NONE;
+
+  if (status === OcrStatus.NONE) {
+    validateFileIsDocument(file);
+  }
+
+  if (status !== OcrStatus.READY && !(await validateLanguage(file.language || 'other'))) {
+    return { status: OcrStatus.UNSUPPORTED_LANGUAGE };
+  }
+
+  return { status, ...(record ? { lastUpdated: record.lastUpdated } : {}) };
+};
+
+const validateTaskIsAdmissible = async (
+  file: EnforcedWithId<FileType>,
+  settingsValues: OcrSettings
+) => {
+  await validateFileIsDocument(file);
+  await validateNotInQueue(file);
+
+  if (!(await validateLanguage(file.language || 'other', settingsValues))) {
+    throw Error('Language not supported');
+  }
+};
+
+class OcrManager {
+  public readonly SERVICE_NAME = 'ocr';
+
+  private ocrTaskManager: TaskManager | null = null;
+
+  start() {
+    this.ocrTaskManager = new TaskManager({
+      serviceName: this.SERVICE_NAME,
+      processResults,
+    });
+  }
+
+  async stop() {
+    await this.ocrTaskManager?.stop();
+    this.ocrTaskManager = null;
+  }
+
+  isReady() {
+    return Boolean(this.ocrTaskManager);
+  }
+
+  private validateIsReady() {
+    if (!this.isReady()) {
+      throw createError('The OCR manager is not ready.', 500);
+    }
+  }
+
+  async addToQueue(file: EnforcedWithId<FileType>) {
+    this.validateIsReady();
+    const settingsValues = await getSettings();
+
+    await validateTaskIsAdmissible(file, settingsValues);
+
+    const fileContent = await readFile(uploadsPath(file.filename));
+    const tenant = tenants.current();
+
+    await request.uploadFile(
+      urljoin(settingsValues.url, 'upload', tenant.name),
+      file.filename,
+      fileContent
+    );
+
+    await this.ocrTaskManager!.startTask({
+      task: this.SERVICE_NAME,
+      tenant: tenant.name,
+      params: {
+        filename: file.filename,
+        language: LANGUAGES_MAP[file.language || 'other'],
```

Comment:
When 'other', is this returning `undefined`? Because the `other` key does not exist on the local hashmap.  Still, replace this with the external one and please look into the `other` issue.

Reply:
It's not an issue. The default is needed here because the FileType has language defined as optional (is it really optional) so we need a default for typescript to work. Any way, if the flow reached this point, file.language will have a valid supported language.
I'll change the strategy to forcing it non-optional inline.